Modules/_io/winconsoleio.c

```diff
@@ -133,6 +133,21 @@ char _PyIO_get_console_type(PyObject *path_or_fd) {
     return m;
 }
 
+static HANDLE py_get_osfhandle(int fd)
```

Comment:
I implemented this, with the full replacement in a separate commit just in case it's frowned upon by a core developer during review.